.github/workflows/pr-realm-js.yml

```diff
@@ -0,0 +1,391 @@
+
+      - name: MSVC Setup
+        if: ${{ runner.os == 'Windows' }}
+        uses: ilammy/msvc-dev-cmd@v1
+
+      - name: Setup Vcpkg
+        if: ${{ runner.os == 'Windows' }}
+        uses: friendlyanon/setup-vcpkg@v1
+        with:
+          path: vendor/realm-core/tools/vcpkg/ports
+          cache-key: vcpkg-windows-${{ matrix.variant.arch }}
+          cache-restore-keys: vcpkg-windows-${{ matrix.variant.arch }}
+
+      # ninja-build is used by default if available and results in faster build times
+      # On linux, electron requires a connected display.  We fake this by giving it a headless environment using xvfb
+      # Relevant issue: https://github.com/juliangruber/browser-run/issues/147
+      - name: Linux Environment setup
+        if: ${{ (matrix.variant.runner == 'ubuntu-20.04') }}
+        run: sudo apt-get install ccache ninja-build xvfb
+
+      - name: ccache
+        uses: hendrikmuhs/ccache-action@v1
+        #TODO: figure out using ccache or equivalent on windows
+        if: ${{ runner.os != 'Windows' }}
+        with:
+          key: ${{ runner.os }}-${{ matrix.variant.os }}-${{ matrix.variant.arch }}
+          max-size: '2.0G'
+
+      - name: Prepend ccache executables to the PATH
+        if: ${{ runner.os != 'Windows' }}
+        run: |
+          echo "/usr/lib/ccache:/usr/local/opt/ccache/libexec" >> $GITHUB_PATH
+
+      - name: Configure ccache
+        if: ${{ runner.os != 'Windows' }}
+        run: |
+          ccache --set-config="compiler_check=content"
+          ccache --show-config
+
+      - name: Install deps
+        run: npm ci --ignore-scripts
+
+      # Install node dependencies
+      - name: Install prebuilds
+        if: steps.cache-prebuild.outputs.cache-hit == 'true'
+        run: npm ci
+
+      # build the c++ library for standard targets
+      - name: Build node
+        if: ${{ (matrix.variant.os != 'ios') && (matrix.variant.os != 'android') }}
+        run: npm run prebuild:node -- --arch=${{matrix.variant.arch}}
+
+      # build the c++ library for IOS
+      - name: Build IOS
+        if: ${{ (matrix.variant.os == 'ios') }}
+        run: npm run prebuild:ios
+
+      # build the c++ library for Android
+      - name: Build Android
+        if: ${{ (matrix.variant.os == 'android') }}
+        run: npm run prebuild:android -- --arch=${{matrix.variant.arch}}
+
+      - name: Install the node dependencies in the test folder
+        working-directory: ./tests
+        run: npm ci --ignore-scripts
+
+      # all tests passed, upload artifacts
+      # the dummy file included is to force preserving paths
+      - name: Upload prebuild artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: realm-js-prebuilds
+          path: |
+            dependencies.list
+            ${{ matrix.variant.artifact-path }}
+
+  publish:
+    runs-on: ubuntu-20.04
```

Comment:
Why can't this be `ubuntu-latest`?

Reply:
This could be brought to a broader discussion.  Should we be pinning the runner versions?  If we always run at latest, then there is a chance things will change without us knowing.

Reply:
Generally, we should aim to always use -latest until we have a good reason. There's no commitment on the GitHub side for how long they'll keep non-latest images around, so it's risky to keep them pinned like this. If an upgrade breaks something, we can always temporarily pin to an earlier version and restore the build while we investigate. But if they remove 20.04, we'd be in a pretty bad spot.

Reply:
It seems prudent to me to fail early.  If our product doesn't work on `-latest` we'll probably leave a lot of users stranded.

Reply:
Ok, I agree with the points made.  I had pinned things due to my own frustrations.  There was a brief moment where `-latest` wasn't consistent between runs.